lib/exceptional_synchrony/event_machine_proxy.rb

```diff
@@ -108,6 +112,14 @@ def ensure_completely_safe(message)
         yield
       end
     end
+
+    def rescue_exceptions_and_ensure_exit(context)
+      yield
+    rescue StandardError
+      # Raise a non-StandardError so that not caught by EM.error_handler.
+      # Expecting rescued exception to be stored in this new exception's cause.
+      raise FatalRunError, "Fatal EventMachine #{context} error"
```

Comment:
Picking up from the Slack conversion: do you think there's any chance that this code will be run in a place that's missing the InvocaMicroservice wrapper that will reveal the `cause`? I think there probably is? If so, maybe hedge our bets at the risk of showing it twice?
```
rescue StandardError => ex
  raise FatalRunError, "Fatal EventMachine #{context} error:\n#{ex.class.name}: #{ex.message}"
```